src/banded/BandedMatrix.jl

```diff
@@ -1007,3 +1007,45 @@ function resize(A::BandedSubBandedMatrix, n::Integer, m::Integer)
     l,u = bandwidths(A)
     _BandedMatrix(reshape(resize!(vec(copy(bandeddata(A))), (l+u+1)*m), l+u+1, m), n, l,u)
 end
+
+function sum(A::BandedMatrix; dims=:)
```

Comment:
Nothing in this implementation uses the fact that its a `BandedMatrix` so we generalise this:
```suggestion
function sum(A::AbstractBandedMatrix; dims=:)
```

In which case the function should probably live in `generic/AbstractBandedMatrix.jl`